Package zapper

sorgtyler opened the pull request:
Because I merged 210 into my development branch, I think the diff is cleanest if I submit a PR into this branch. 

### New class: 
- uc_delete_packages

### Code modifications:
- I modified uc_delete_conditions to more closely resemble uc_delete_packages' new behavior, specifically progress bar update math and clearing the dictionaries between subsequent runs. 
- I also resized the parent form and hid the not completely functional plot deleter.

**Don't merge until after today's dev call in case I need to make more changes.**

sorgtyler commented:
Something to discuss on the call: Are (new?) CompactMDB errors reproducible on FS machines? I keep seeing an error saying the file is open by another user. BioSum keeps running, but the compact operations repeatedly throw errors.

lbross commented:
I'm not clear on if I should merge this pull request. Do you think there will be more coming within the next week?

sorgtyler commented:
@lbross  I moved the log file creation to a finally block at the end of the deletions and changed the format of the logs a little. I think this is a good point to merge the code in. Since the debug log writes lines one at a time where I would incrementally write to the user's requested log, I think I should clarify the differences in what is being printed to each log.
